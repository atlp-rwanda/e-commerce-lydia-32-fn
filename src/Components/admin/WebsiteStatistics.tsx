import React from 'react'
import { Line } from 'react-chartjs-2';
import { Chart as ChartJS, ArcElement, Tooltip, Legend, CategoryScale, LinearScale, PointElement, LineElement, Title } from 'chart.js';

ChartJS.register(ArcElement, Tooltip, Legend, CategoryScale, LinearScale, PointElement, LineElement, Title);

const WebsiteStatistics: React.FC = () => {
    const lineData = {
        labels: ['January', 'February', 'March', 'April', 'May', 'June', 'July'],
        datasets: [
          {
            label: 'Website Traffic',
            data: [65, 59, 80, 81, 56, 55, 40],
            fill: false,
            borderColor: 'rgba(75,192,192,1)',
          },
        ],
      };
    
      return (
        <div className="bg-white p-4 shadow rounded">
          <h2 className="text-lg font-bold mb-4">Website Statistics</h2>
          <Line data={lineData} />
        </div>
      );
}

export default WebsiteStatistics
