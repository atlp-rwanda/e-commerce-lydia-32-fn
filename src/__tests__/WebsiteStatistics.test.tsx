import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { vi } from 'vitest';
import WebsiteStatistics from '../Components/admin/WebsiteStatistics';
import { Line } from 'react-chartjs-2';

// Mock the Line component from react-chartjs-2
vi.mock('react-chartjs-2', () => ({
  Line: vi.fn(() => <canvas data-testid="chart"></canvas>),
}));

describe('WebsiteStatistics Component', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  it('should render without crashing', () => {
    render(<WebsiteStatistics />);
    
    const heading = screen.getByText('Website Statistics');
    expect(heading).toBeInTheDocument();

    const chart = screen.getByTestId('chart');
    expect(chart).toBeInTheDocument();
  });

  it('should pass correct data to Line chart', () => {
    const expectedData = {
      labels: ['January', 'February', 'March', 'April', 'May', 'June', 'July'],
      datasets: [
        {
          label: 'Website Traffic',
          data: [65, 59, 80, 81, 56, 55, 40],
          fill: false,
          borderColor: 'rgba(75,192,192,1)',
        },
      ],
    };

    render(<WebsiteStatistics />);
    
    expect(Line).toHaveBeenCalledWith(
      expect.objectContaining({
        data: expectedData,
      }),
      {}
    );
  });

  it('should match snapshot', () => {
    const { container } = render(<WebsiteStatistics />);
    expect(container).toMatchSnapshot();
  });
});
