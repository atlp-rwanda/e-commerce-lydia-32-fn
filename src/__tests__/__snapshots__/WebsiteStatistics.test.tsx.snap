// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WebsiteStatistics Component > should match snapshot 1`] = `
<div>
  <div
    class="bg-white p-4 shadow rounded"
  >
    <h2
      class="text-lg font-bold mb-4"
    >
      Website Statistics
    </h2>
    <canvas
      data-testid="chart"
    />
  </div>
</div>
`;
